packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -538,8 +542,13 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
             ...sessionPoolOptions,
             log,
         };
-        if (this.retryOnBlocked && !sessionPoolOptions.blockedStatusCodes) {
-            this.sessionPoolOptions.blockedStatusCodes = [];
+        if (this.retryOnBlocked) {
+            if (!sessionPoolOptions.blockedStatusCodes) {
```

Comment:
i guess we should also check for empty array, we dont want to print warnings for that case